pkg/user/manager/rest/rest.go

```diff
@@ -0,0 +1,341 @@
+	err := m.renewAPIToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("%s/Identity/?filter=id:%s&field=upn&field=primaryAccountEmail&field=displayName", m.conf.APIBaseURL, uid.OpaqueId)
+
+	httpClient := rhttp.GetHTTPClient(ctx)
+	httpReq, err := rhttp.NewRequest(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	httpReq.Header.Set("Authorization", "Bearer "+m.apiToken)
+
+	httpRes, err := httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(httpRes.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	userData, ok := result["data"].([]interface{})[0].(map[string]interface{})
+	if !ok {
+		return nil, errors.New("rest: error in type assertion")
+	}
+
+	userGroups, err := m.GetUserGroups(ctx, uid)
+	if err != nil {
+		return nil, err
+	}
+
+	return &userpb.User{
+		Id:          uid,
+		Username:    userData["upn"].(string),
+		Mail:        userData["primaryAccountEmail"].(string),
+		DisplayName: userData["displayName"].(string),
+		Groups:      userGroups,
+	}, nil
+
+}
+
+func (m *manager) findUsersByFilter(ctx context.Context, url string) ([]*userpb.User, error) {
+
+	err := m.renewAPIToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := rhttp.GetHTTPClient(ctx)
+	httpReq, err := rhttp.NewRequest(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	httpReq.Header.Set("Authorization", "Bearer "+m.apiToken)
+
+	httpRes, err := httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(httpRes.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	userData, ok := result["data"].([]interface{})
+	if !ok {
+		return nil, errors.New("rest: error in type assertion")
+	}
+
+	users := []*userpb.User{}
+
+	for _, usr := range userData {
+		usrInfo, ok := usr.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("rest: error in type assertion")
+		}
+
+		uid := &userpb.UserId{
+			OpaqueId: usrInfo["id"].(string),
+			Idp:      m.conf.IDProvider,
+		}
+
+		userGroups, err := m.GetUserGroups(ctx, uid)
+		if err != nil {
+			return nil, err
+		}
+		users = append(users, &userpb.User{
+			Id:          uid,
+			Username:    usrInfo["upn"].(string),
+			Mail:        usrInfo["primaryAccountEmail"].(string),
+			DisplayName: usrInfo["displayName"].(string),
+			Groups:      userGroups,
+		})
+	}
+
+	return users, nil
+}
+
+func (m *manager) FindUsers(ctx context.Context, query string) ([]*userpb.User, error) {
+
+	filters := []string{"upn", "displayName", "primaryAccountEmail"}
+	users := []*userpb.User{}
+
+	for _, f := range filters {
+		url := fmt.Sprintf("%s/Identity/?filter=%s:contains:%s&field=id&field=upn&field=primaryAccountEmail&field=displayName", m.conf.APIBaseURL, f, query)
```

Comment:
%s can contain arbitrary user input I imagine, we  need to whitelist so we only accept expected characters to avoid an injection attack.

Reply:
Added regexs for email and names.